crypto/crypto/src/ed25519.rs

```diff
@@ -214,6 +218,19 @@ impl SigningKey for Ed25519PrivateKey {
     type VerifyingKeyMaterial = Ed25519PublicKey;
     type SignatureMaterial = Ed25519Signature;
 
+    fn sign<T: CryptoHash + Serialize>(
```

Comment:
@huitseeker what happens if one invokes this accidentally with a bytes, actually Vec object, after they serialized the original struct.
It should be clear to devs that they don’t serialize the struct outside and this function takes as input the original object type. Maybe add some docs clarifying that.

Reply:
The docstring is on the trait.